cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -124,33 +123,57 @@ def test_gce_preemptible_provision(appliance, provider, instance_args, soft_asse
     soft_assert(instance.exists_on_provider, "Instance wasn't provisioned successfully")
 
 
-@pytest.mark.manual
-@pytest.mark.meta(coverage=[1676910])
-def test_provision_approval_deny():
-    """ Test whether the approver can deny the request.
+def post_approval(smtp_test, provision_request, vm_type, requester, provider, vm_names):
+    # requester includes the trailing space
+    approved_subject = normalize_text("your {} request was approved".format(vm_type))
+    approved_from = normalize_text("{} request from {}was approved".format(vm_type, requester))
 
-    Prerequisities:
-        * A provider that can provision.
+    wait_for_messages_with_subjects(smtp_test, [approved_subject, approved_from], num_sec=90)
 
-    Steps:
-        * Create a provisioning request that does not get automatically approved (eg. ``num_vms``
-            bigger than 1)
-        * Deny the created request
+    smtp_test.clear_database()
 
-    Polarion:
-        assignee: jhenner
-        caseimportance: high
-        casecomponent: Provisioning
-        initialEstimate: 1/3h
+    # Wait for the VM to appear on the provider backend before proceeding to ensure proper cleanup
+    logger.info('Waiting for vms %s to appear on provider %s', ", ".join(vm_names), provider.key)
+    wait_for(
+        lambda: all(map(provider.mgmt.does_vm_exist, vm_names)),
+        handle_exception=True,
+        num_sec=600
+    )
+
+    provision_request.wait_for_request(method='ui')
+    msg = "Provisioning failed with the message {}".format(provision_request.row.last_message.text)
+    assert provision_request.is_succeeded(method='ui'), msg
+
+    # account for multiple vms, specific names
+    completed_subjects = [
+        normalize_text("your {} request has completed vm name {}".format(vm_type, name))
+        for name in vm_names
+    ]
+    wait_for_messages_with_subjects(smtp_test, completed_subjects, num_sec=90)
+
+
+def wait_for_messages_with_subjects(smtp_test, expected_subjects, num_sec):
+    """ This waits for all the expected subjects to be present the list of received
+    mails with partial match.
     """
-    pass
+    def check_subjects():
```

Comment:
_Optional_ since this is an internal method
```suggestion
    def _check_subjects():
```

Reply:
yeah it is internal -- not really accessible from outside and I think that makes clear it is "private" in Java terms. So I am not sure what the underscore helps with, but I have no problems to change it.

Reply:
It is just preference for me, which is why I made it _optional_. Though in python there are no truly private methods, I still think it is good practice to mark internal/private methods with the pre-pended `_`